src/main/java/ratpack/zipkin/internal/TracingPropagationExecInitializer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package ratpack.zipkin.internal;
+
+import brave.http.HttpTracing;
+import brave.propagation.TraceContext;
+import ratpack.exec.ExecInitializer;
+import ratpack.exec.Execution;
+
+/**
+ * ExecInitializer that will propagate the tracing context into any new execution created.
+ */
+public class TracingPropagationExecInitializer implements ExecInitializer {
```

Comment:
I wonder if this could be a static inner class of `RatpackCurrentTraceContext`.  Mostly suggesting this so we can continue keeping `TraceContextHolder` private.  Honestly I'm not 100% sure that will work but just trying to see if we can not leak access to the otherwise private class.

Reply:
I don't know if we should keep `TraceContextHolder` private it makes it very painful to get in and mess with things which sometimes is needed. 

If you want to try and keep everything locked in I am happy to do that.

Reply:
Making things harder to mess with is precisely why I suggest keeping it private.  It's not to say that I'm suggesting making folks life harder but rather protecting the project from future difficulties.  As soon as its accessible then it becomes part of the API and not something we can easily make breaking changes to in the future.  Marking this as `protected` is certainly an option if others feel it is the right course of action as well :)